commit caa5ead18c00c8924f5b03d6495148e51e8f3cdf
Author: Dawit Alemayehu <adawit@kde.org>
Date:   Fri Mar 4 18:58:28 2011 -0500

    Fix the issue of HTTPS over proxy not working properly when connecting to two or more secure sites.
    
    BUG:155707
    FIXED-IN:4.6.2

diff --git a/kioslave/http/http.cpp b/kioslave/http/http.cpp
index 534de3b..25cb1cc 100644
--- a/kioslave/http/http.cpp
+++ b/kioslave/http/http.cpp
@@ -2089,7 +2089,7 @@ bool HTTPProtocol::httpShouldCloseConnection()
   // TODO compare current proxy state against proxy needs of next request,
   // *when* we actually have variable proxy settings!
 
-  if (isValidProxy(m_request.proxyUrl))  {
+  if (isHttpProxy(m_request.proxyUrl) && !isAutoSsl())  {
       return !isCompatibleNextUrl(m_server.proxyUrl, m_request.proxyUrl);
   }
   return !isCompatibleNextUrl(m_server.url, m_request.url);
